### ISSUE TITLE
planning

### ISSUE THREAD

**Issue by pguth** [thingylabs/crypto-thingy#1]
@kkaufmann proposed sticking to standard file formats. More specifically [7zip](https://www.npmjs.com/package/node-7z).

**Comment by queicherius** [comment]
Here my thoughts when I first thought about it:

- Minimalist GUI: basically just a thing to choose / drop files, display them?, a progress bar, a field for a password and encrypt/decrypt buttons
- Drag and Drop files

I think that's literally it for a first version. The sharing of urls via webtorrent would make a good improvement for a v2. 

--

> @kkaufmann proposed sticking to standard file formats. More specifically 7zip. Or via a system call: 7z a -tzip -mx=0 -mem=AES256 -p<password> <archive-name> <file>

That would include 7z as a dependency, whereas just using the built in crypto module with `aes-256-ctr` would not. Not sure. About this. Which AES mode does 7z even use?

**Comment by queicherius** [comment]
Brain dump wireframe:

![](http://i.imgur.com/Ik0Q1Tt.png)

It would also be possible to do the steps clearer separated (with headings or different screens)

1. Put in the files
2. Enter your password
3. Choose encrypt or decrypt

**Comment by pguth** [comment]
How 'bout even simpler, postponing the password dialogue. More like this:

<img width="532" alt="screen shot 2016-07-15 at 15 17 11" src="https://cloud.githubusercontent.com/assets/85259/16875369/6453f878-4a9f-11e6-9345-3015a8eecb5c.png">

**Comment by queicherius** [comment]
Yep, that's pretty much what I meant with the different screens. Then the Password + Buttons on the second screen? Would you get to the second screen automatically? How does the WebTorrent tool do it, maybe we can learn from that :)
